### PR TITLE
Allow user to install jax-head,jaxlib-nightly,libtpu-nightly or customized libtpu

### PR DESCRIPTION
**Pull request by tonyjohnchen**
Allow user to install jax-head,jaxlib-nightly,libtpu-nightly.
Allow user to install customized libtpu from GCS bucket.

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -14,4 +14,31 @@
...
 # limitations under the License.
 
-pip3 install -r requirements.txt -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
\ No newline at end of file
+# Uninstall existing JAX and JAXlib
+pip3 show jax && pip3 uninstall -y jax 
+pip3 show jaxlib && pip3 uninstall -y jaxlib
+
+# Install dependencies from requirements.txt
+pip3 install -r requirements.txt -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
```
Comment (rwitten) [inline setup.sh:22]: Not sure I understand this. We install `requirements.txt` -- but doesn't taht still contain `jax[tpu]`? Shouldn't we remove `jax[tpu]` from requirements.txt and only install it if necessary?

File: setup.sh [hunk 2]
```diff
@@ -14,4 +14,31 @@
...
+done
+
+if [[ $MODE == "nightly" ]]; then
+    echo "Installing jax-head, jaxlib-nightly, and libtpu-nightly"
+    # Install JAX from GitHub head
+    pip3 install git+https://github.com/google/jax
+    # Install jaxlib-nightly
+    pip3 install --pre -U jaxlib -f https://storage.googleapis.com/jax-releases/jaxlib_nightly_releases.html
+    # Install libtpu-nightly
+    pip3 install libtpu-nightly -f https://storage.googleapis.com/jax-releases/libtpu_releases.html -U --pre
```
Comment (rwitten) [inline setup.sh:37]: We shouldn't install a `libtpu-nightly` if we're given a gcs path, that seems confusing.

File: setup.sh [hunk 3]
```diff
@@ -14,4 +14,31 @@
...
+fi
+
+if [[ -n "$LIBTPU_GCS_PATH" ]]; then
+    sudo pip3 install -U crcmod
+    echo "Copying libtpu.so from $LIBTPU_GCS_PATH to /lib/libtpu.so"
+    sudo gsutil cp "$LIBTPU_GCS_PATH" /lib/libtpu.so
+fi
```
Comment (rwitten) [inline setup.sh:44]: Having now read to this point, I recommend the following order.

(1) Uninstall jax, jaxlib and libtpu and delete `/lib/libtpu.so`
(2) Install jax, jaxlib and libtpu
- if `stable`, via jax[tpu]. Assert the user doesn't pass in a libtpu separately.
- if `nightly`, install jax and jaxlib as you do. Install libtpu from either GCS or libtpu nightly as appropriate. 
(3) install the rest of requirements.

**Review by rwitten — CHANGES_REQUESTED** [review]
Thanks for pulling this together, I have some ideas

File: setup.sh (hunk 3 above)
Comment (tonyjohnchen) [reply setup.sh:60]: Just made the change based on your comments.

File: setup.sh [hunk 4]
```diff
@@ -14,4 +14,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-pip3 install -r requirements.txt -f https://storage.googleapis.com/jax-releases/libtpu_releases.html
\ No newline at end of file
+# Uninstall existing JAX and JAXlib
+pip3 show jax && pip3 uninstall -y jax 
+pip3 show jaxlib && pip3 uninstall -y jaxlib
+
```
Comment (rwitten) [inline setup.sh:20]: shouldn't we uninstall libtpu?

**Review by gobbleturk — APPROVED** [review]

File: setup.sh [hunk 5]
```diff
@@ -14,4 +14,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
```
Comment (rwitten) [inline setup.sh:16]: Can we `set -e` this script? So if there is an error in the code, it propagates to the top level error status?

**Review by rwitten — CHANGES_REQUESTED** [review]
Two tiny things!

File: setup.sh (hunk 4 above)
Comment (tonyjohnchen) [reply setup.sh]: Done

File: setup.sh (hunk 5 above)
Comment (tonyjohnchen) [reply setup.sh:16]: Done

**Review by rwitten — APPROVED** [review]

**Pull request merged**